test/unit/components/withIntl.js

```diff
@@ -2,144 +2,149 @@ import expect, {spyOn} from 'expect';
-        expect(withIntl(Wrapped).WrappedComponent).toBe(Wrapped);
+    it("includes `WrappedComponent`'s `displayName`", () => {
+      Wrapped.displayName = 'Foo';
+      expect(withIntl(Wrapped).displayName).toBe('withIntl(Foo)');
     });
-
-    it('hoists non-react statics',() => {
-        expect(withIntl(Wrapped).someNonReactStatic.foo).toBe(true)
-    })
-
-    describe('displayName', () => {
-        it('is descriptive by default', () => {
-            expect(withIntl(() => null).displayName).toBe('withIntl(Component)');
-        });
-
-        it('includes `WrappedComponent`\'s `displayName`', () => {
-            Wrapped.displayName = 'Foo';
-            expect(withIntl(Wrapped).displayName).toBe('withIntl(Foo)');
-        });
+  });
+
+  it('throws when <IntlProvider> is missing from ancestry', () => {
+    const Injected = withIntl(Wrapped);
+
+    const consoleError = spyOn(console, 'error'); // surpress console error from JSDom
+    expect(() => (rendered = mount(<Injected />))).toThrow(
+      '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
+    );
+    consoleError.restore();
+  });
+
+  it('renders <WrappedComponent> with `intl` prop', () => {
+    const Injected = withIntl(Wrapped);
+
+    rendered = mountWithProvider(<Injected />);
+    const wrappedComponent = rendered.find(Wrapped);
+    // React 16 renders different in the wrapper
+    const intlProvider = rendered.find(IntlProvider).childAt(0);
+
+    expect(wrappedComponent.prop('intl')).toBe(
+      intlProvider.instance().getContext()
+    );
+  });
+
+  it('propagates all props to <WrappedComponent>', () => {
+    const Injected = withIntl(Wrapped);
+    const props = {
+      foo: 'bar',
+    };
+
+    rendered = mountWithProvider(<Injected {...props} />);
+    const wrappedComponent = rendered.find(Wrapped);
+
+    Object.keys(props).forEach(key => {
+      expect(wrappedComponent.prop(key)).toBe(props[key]);
     });
+  });
 
-    it('throws when <IntlProvider> is missing from ancestry', () => {
-        const Injected = withIntl(Wrapped);
+  describe('options', () => {
+    describe('intlPropName', () => {
+      it("sets <WrappedComponent>'s `props[intlPropName]` to `context.intl`", () => {
+        const propName = 'myIntl';
+        Wrapped.propTypes = {
+          [propName]: intlShape.isRequired,
+        };
+        const Injected = withIntl(Wrapped, {
+          intlPropName: propName,
+        });
 
-        const consoleError = spyOn(console, 'error'); // surpress console error from JSDom
-        expect(() => rendered = mount(<Injected />)).toThrow(
-            '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
+        rendered = mountWithProvider(<Injected />);
+        const wrapped = rendered.find(Wrapped);
+        // React 16 renders differently
+        const intlProvider = React.version.startsWith('16')
```

Comment:
This can be removed due to version 3 of `react-intl` only supporting `react@16.3.1` and above - my bad for having missed it!